File: 1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */
package main

// @lc code=start
func twoSum(nums []int, target int) []int {
	if len(nums) <= 2 {
		return []int{0, 1}
	}
	var numsMap = make(map[int]int)
	for i, v := range nums {
		temp := target - v
		if index, ok := numsMap[temp]; ok && index != i {
			return []int{index, i}
		} else {
			numsMap[v] = i
		}
	}
	return []int{0, 0}
}

// @lc code=end

File: 1.两数之和.cs
/*
 * @lc app=leetcode.cn id=1 lang=csharp
 *
 * [1] 两数之和
 */

// @lc code=start
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        int[] res = new int[2]{0,1};
        Dictionary<int, int> dir = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++)
        {
            int temp = target - nums[i];
            if (dir.ContainsKey(temp) && dir[temp] != i)
            {
                res = new int[2] { dir[temp], i };
                return res;
            }
            else
            {
                dir[nums[i]] = i;
            }
        }
        return res;
    }
}
// @lc code=end